### PR TITLE
Add missing driver numbers

### PR DESCRIPTION
**Pull request by adkinsjd**
### Pull Request Overview

Adds the driver numbers that are documented but not included in the drivers themselves.

### Testing Strategy

N/A

### TODO or Help Wanted

N/A

### Documentation Updated

- [x] Kernel: Updated the relevant files in `/docs`, or no updates are required.
- [x] Userland: Added/updated the application README, if needed.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

File: capsules/src/i2c_master_slave_driver.rs [hunk 1]
```diff
@@ -18,6 +18,9 @@ use kernel::ReturnCode;
 use kernel::common::take_cell::{MapCell, TakeCell};
 use kernel::hil;
 
+// Syscall driver number.
+pub const DRIVER_NUM: usize = 0x20003;
```
Comment (ppannuto) [inline capsules/src/i2c_master_slave_driver.rs]: Actually, on second thought, I'm not sure that this one is right. The docs say
> doc/syscalls/README.md:|   | 0x20003       | I2C Master       | Raw I2C Master interface                   |
but this is a mixed master/slave thing.

Can you pull this one out? #761 gives this driver a temporary number while we sort out what exactly we want to do here.

Comment (adkinsjd) [reply capsules/src/i2c_master_slave_driver.rs]: You're right - I missed that, and I can pull it out

Neither of the documented I2C drivers are used anywhere in tock though, so we should think about how to fix this, either by separating the current driver into master/slave pieces or committing to them being combined.

**Comment by adkinsjd** [comment]
I just realized that some of these driver numbers are not fixed in their userland libraries either, so I'm going to update those as well.

**Comment by niklasad1** [comment]
@adkinsjd This is awesome but please use ```rustdoc comments``` for public items i.e, ```///``` instead of ```//```

File: capsules/src/ltc294x.rs [hunk 2]
```diff
@@ -50,6 +50,9 @@ use kernel::common::take_cell::TakeCell;
 use kernel::hil::gpio;
 use kernel::hil::i2c;
 
+// Syscall driver number.
```
Comment (ppannuto) [inline capsules/src/ltc294x.rs]: I think you missed one :)

File: userland/libtock/internal/nonvolatile_storage.h [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 #include "tock.h"
 
-#define DRIVER_NUM_NONVOLATILE_STORAGE 27
+#define DRIVER_NUM_NONVOLATILE_STORAGE 0x00050001
```
Comment (niklasad1) [inline userland/libtock/internal/nonvolatile_storage.h]: Un-necessary zeros

**Pull request merged**